loader/checkpoint.go

```diff
@@ -80,13 +80,14 @@ type RemoteCheckPoint struct {
 }
 
 func newRemoteCheckPoint(tctx *tcontext.Context, cfg *config.SubTaskConfig, id string) (CheckPoint, error) {
-	db, dbConns, err := createConns(tctx, cfg, 1)
+	tctx2, cancel := tctx.WithTimeout(defaultDBContextTimeout)
+	defer cancel()
+
+	db, dbConns, err := createConns(tctx2, cfg, 1)
```

Comment:
does context with timeout affect `resetBaseConnFn` in `createConns`?

Reply:
No, `tctx` in `resetBaseConnFn` is passed in (it's a function argument) when reseting connetion like in `func (conn *DBConn) resetConn(tctx *tcontext.Context) error`.

this timeout context only used to `GetBaseConn` and log something.